### PR TITLE
Rename ChannelTypeBuilder to GenericChannelTypeBuilder and fix packages

### PR DESCRIPTION
**Pull request by triller-telekom**
See https://github.com/eclipse/smarthome/pull/5866#discussion_r202479275

Signed-off-by: Stefan Triller <stefan.triller@telekom.de>

### CODE REVIEW

**Comment by triller-telekom** [comment]
Since `default` methods need and implementation to be actually called, I changed it to have static methods on the `ChannelTypeBuilder` interface so I could drop the `Generic` one.

File: bundles/core/org.eclipse.smarthome.core.thing.test/src/test/java/org/eclipse/smarthome/core/thing/type/ChannelTypeBuilderTest.java [hunk 1]
```diff
@@ -30,7 +30,7 @@
 import org.junit.Test;
 
 /**
- * Tests the {@link ChannelTypeBuilder}.
+ * Tests the {@link GenericChannelTypeBuilder}.
```
Comment (htreu) [inline bundles/core/org.eclipse.smarthome.core.thing.test/src/test/java/org/eclipse/smarthome/core/thing/type/ChannelTypeBuilderTest.java]: some references to the intermediate `GenericChannelTypeBuilder` which are obsolete now.

Comment (triller-telekom) [reply bundles/core/org.eclipse.smarthome.core.thing.test/src/test/java/org/eclipse/smarthome/core/thing/type/ChannelTypeBuilderTest.java]: Good catch. Unfortunately deleting a file is no "refactor" action in Eclipse :(

**Comment by wborn** [comment]
Shouldn't the deprecations be updated because of this change?

**Comment by triller-telekom** [comment]
@wborn You are right, there was a second constructor in `ChannelType` I have now set both to deprecated. Thanks for noticing!

**Comment by cweitkamp** [comment]
One question: We now have deprecated all constructors for the `ChannelType` - which obviously makes sense. I skipped the second one in #5881 because we did not use the deprecation on the remaining `ThingType` constructor. With this PR I have to ask why? Is there a specific reason to leave the most parameterized one without the annotation? If not: Shall we deprecate all of them even in `ThingType` and `ChannelGroupType`?

**Pull request merged**